src/ent/board.h

```diff
@@ -182,76 +473,182 @@ static inline bool board_are_all_adjacent_squares_empty(const Board *board,
+    }
+  }
 }
 
-static inline bool board_get_transposed(const Board *board) {
-  return board->transposed;
+static inline void board_update_all_anchors(Board *board) {
+  if (board->tiles_played > 0) {
+    for (int i = 0; i < BOARD_DIM; i++) {
+      for (int j = 0; j < BOARD_DIM; j++) {
+        board_update_anchors(board, i, j);
+      }
+    }
+  } else {
+    for (int i = 0; i < BOARD_DIM; i++) {
+      for (int j = 0; j < BOARD_DIM; j++) {
+        board_reset_anchor(board, i, j, 0);
+        board_reset_anchor(board, i, j, 1);
+      }
+    }
+    board_reset_number_of_anchor_rows(board);
+    board_set_anchor(board, BOARD_DIM / 2, BOARD_DIM / 2, 0, true);
+  }
 }
 
-static inline void board_transpose(Board *board) {
-  board->transposed = !board->transposed;
-}
+static inline void board_reset(Board *board) {
+  // The transposed field must be set to 0 here because
+  // it is used to calculate the index for board_set_letter.
+  board->tiles_played = 0;
+  board->transposed = 0;
 
-static inline void board_set_transposed(Board *board, bool transposed) {
-  board->transposed = transposed;
-}
+  for (int row = 0; row < BOARD_DIM; row++) {
+    for (int col = 0; col < BOARD_DIM; col++) {
+      board_set_letter(board, row, col, ALPHABET_EMPTY_SQUARE_MARKER);
+      board_reset_is_cross_word(board, row, col, BOARD_HORIZONTAL_DIRECTION);
+      board_reset_is_cross_word(board, row, col, BOARD_VERTICAL_DIRECTION);
+    }
+  }
 
-static inline bool board_matches_dir(const Board *board, int dir) {
-  return (board_is_dir_vertical(dir) && board_get_transposed(board)) ||
-         (!board_is_dir_vertical(dir) && !board_get_transposed(board));
+  board_set_all_crosses(board);
+  board_reset_all_cross_scores(board);
+  board_update_all_anchors(board);
+}
+
+static inline void board_init_bonus_squares(Board *b) {
+  int i = 0;
+  for (int row = 0; row < BOARD_DIM; row++) {
+    for (int col = 0; col < BOARD_DIM; col++) {
+      uint8_t bonus_value;
+      char bonus_square = CROSSWORD_GAME_BOARD[i++];
+      if (bonus_square == BONUS_TRIPLE_WORD_SCORE) {
+        bonus_value = 0x31;
+      } else if (bonus_square == BONUS_DOUBLE_WORD_SCORE) {
+        bonus_value = 0x21;
+      } else if (bonus_square == BONUS_DOUBLE_LETTER_SCORE) {
+        bonus_value = 0x12;
+      } else if (bonus_square == BONUS_TRIPLE_LETTER_SCORE) {
+        bonus_value = 0x13;
+      } else {
+        bonus_value = 0x11;
+      }
+      board_set_bonus_square(b, row, col, bonus_value);
+    }
+  }
 }
 
-static inline int board_get_tiles_played(const Board *board) {
-  return board->tiles_played;
+static inline Board *board_create() {
+  Board *board = malloc_or_die(sizeof(Board));
+  board_reset(board);
+  board_init_bonus_squares(board);
+  return board;
 }
 
-static inline bool board_get_path_is_valid(const Board *board) {
-  return board->path_is_valid;
+static inline void board_copy(Board *dst, const Board *src) {
+  memory_copy(dst, src, sizeof(Board));
```

Comment:
make memory_copy static inline

Reply:
Forgot I did this in a commit I overwrote. Should be fixed now.